Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/Instance/InstanceDomGeneratorInterface.h

```diff
@@ -21,9 +21,11 @@ namespace AzToolsFramework
             AZ_RTTI(InstanceDomGeneratorInterface, "{269DE807-64B2-4157-93B0-BEDA4133C9A0}");
             virtual ~InstanceDomGeneratorInterface() = default;
 
-            // Given a pointer to an instance and a reference to a dom, update the dom so that it represents the instance,
-            // and return if the dom creation succeeded or not.
-            virtual bool GenerateInstanceDom(const Instance* instance, PrefabDom& instanceDom) = 0;
+            //! Generates an instance DOM that represents a given instance object.
+            //! @param instanceDom The output instance DOM that will be modified.
```

Comment:
nit: using @param[out] for output variables is preferred to distinguish them from the common input parameters. If it's used both as an input and out parameter use, @param[in,out]